src/UI/Components/Diagram/components/RightSidebar.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { useContext, useEffect, useState } from "react";
+import {
+  Button,
+  Flex,
+  FlexItem,
+  TextContent,
+  Title,
+} from "@patternfly/react-core";
+import styled from "styled-components";
+import { Field, InstanceAttributeModel, ServiceModel } from "@/Core";
+import { sanitizeAttributes } from "@/Data";
+import { words } from "@/UI/words";
+import { CanvasContext, InstanceComposerContext } from "../Context/Context";
+import { updateServiceOrderItems } from "../helpers";
+import { ActionEnum, EmbeddedEventEnum } from "../interfaces";
+import { EntityForm } from "./EntityForm";
+
+/**
+ * `RightSidebar` is a React functional component that renders a sidebar for editing and removing entities.
+ * The sidebar displays the details of the selected entity and provides options to edit or remove the entity.
+ * The state of the sidebar is updated based on the selected entity and the user's interactions with the sidebar and/or composer's canvas.
+ * When the user submits the edit form, the `onSave` callback is called with the updated attributes and the form state.
+ * The user can also remove the entity by clicking the remove button, which triggers the `action:delete` event on the entity.
+ *
+ * The removal of the entity has different end result based on the type of the entity:
+ * - Core entity cannot be removed.
+ * - Embedded entities are removed from the canvas(if service model allows it), and will be erased from the service instance.
+ * - Inter-service relation entities are removed from the canvas(if service model allows it), but won't be deleted from the environment.
+ *
+ *
+ * @returns {React.FC} The RightSidebar component.
+ */
+export const RightSidebar: React.FC = () => {
+  const { cellToEdit, diagramHandlers, setServiceOrderItems, stencilState } =
+    useContext(CanvasContext);
+  const { mainService } = useContext(InstanceComposerContext);
+  const [description, setDescription] = useState<string | null>(null);
+  const [isAllowedToRemove, setIsAllowedToRemove] = useState(false);
```

Comment:
`isRemovable`, or `isAllowedToBeRemoved`
Because is allowed to remove means they can remove something, while you mean they can be removed. I didn't make the suggestion since this is a change that will impact the rest of the code.